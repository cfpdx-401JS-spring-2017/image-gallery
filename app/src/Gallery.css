
.YoungDivvy {
    height: 100%;
    width: 100vw;
    background-color: #84A98C;
    font-size: 3em;
    position: absolute;
    top: 0;
    left:0;
    z-index: 2;
    display: flex;
    justify-content: center;
    align-items: center
}

.mainButtons {
    text-decoration: none;
    background-color: #84A98C;
    color: #2F3E46;
    border: 5px solid #2F3E46;
    border-radius: 15px;
    margin: 2vh 2vh;
    width: 50vw;
    height: 20vh;
    font-size: 3em;
}

img {
    width: 50vw;
    max-height: 60vh;
    border: 5px solid white;
    border-radius: 10px;
}

.arrows {
    border-radius: 100%;
    border: 5px solid white;
    width: 10vh;
    height: 10vh;
    padding-top: 2vh;
    margin: 3vh;
    z-index: 4;
 
}

h5 {
    font-size: .5em;
}

.ImageTitle {
    font-size: .7em;
}

.exitButton  {
    position: absolute;
    top: 0vh;
    left: 90vw;
}