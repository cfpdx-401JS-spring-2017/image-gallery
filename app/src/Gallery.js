import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Gallery extends Component {

  constructor(props) {
    super(props);
    
    this.state = {
      index: 0
    }

    this.handleLeftButton = this.handleLeftButton.bind(this);
    this.handleRightButton = this.handleRightButton.bind(this);
  }

  static propTypes = {
    data: PropTypes.array.isRequired
  }

  handleLeftButton() {
    let decrementIndex = this.state.index;
    decrementIndex--;
    const index = decrementIndex === -1 ? this.props.data.length - 1 : decrementIndex;
    this.setState({ index });
  }

  handleRightButton() {
    let incrementIndex = this.state.index;
    incrementIndex++;
    const index = incrementIndex === this.props.data.length ? 0 : incrementIndex;
    this.setState({ index });
  }

  render() {
    let { data } = this.props;

    if (!data.length) return (
      <div>
        <img src='https://media.giphy.com/media/3oEjI6SIIHBdRxXI40/giphy.gif' alt='loading'/>
      </div>
    );

    let currentImage = data[this.state.index];
    let { title, description, url } = currentImage;

    return (
      <div className='Gallery'>
        <div className='button-container'>
          <button onClick={this.handleLeftButton}>&larr;</button>
          <button onClick={this.handleRightButton}>&rarr;</button>
        </div>
        <div className='Gallery-view'>
          <h2>{title}</h2>
          <p>{description}</p>
          <img src={url} alt={title} className='Gallery-image' />
        </div>
      </div>
    );
  }
}

export default Gallery;