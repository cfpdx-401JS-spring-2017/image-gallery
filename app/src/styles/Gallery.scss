.gallery-btn {
  display: inline-block;
  width: 8%;
  text-align: center;
}