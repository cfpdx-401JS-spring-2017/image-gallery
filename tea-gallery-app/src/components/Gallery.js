import React from 'react';
import PropTypes from 'prop-types';
import GalleryItem from './GalleryItem';

Gallery.PropTypes = {
  images: PropTypes.array,
  onDeleteImage: PropTypes.func.required
};

export default function Gallery(props) {
  return (
    <GalleryItem
      images={props.images}
      onDeleteImage={props.onDeleteImage} 
    />
  );
}