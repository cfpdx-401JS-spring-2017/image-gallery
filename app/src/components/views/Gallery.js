import React, { Component } from 'react';
import PropTypes from 'prop-types';
import GalleryImage from './GalleryImage';
import '../../styles/Gallery.css';

export default class Gallery extends Component {

  static propTypes = {
    poms: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
      description: PropTypes.string,
      url: PropTypes.string.isRequired
    }))
  };

  constructor(props) {
    super(props);

    this.state = {
      index: 0
    };
  }

  changePom(increment) {
    const poms = this.props.poms;
    const pom = this.props.poms[this.state.index];
    let index = this.state.index;

    index += increment;
    if (index === poms.length) {
      index = 0;
    } if (index === -1 || isNaN(index)) {
      index = poms.length - 1;
    }
    this.setState({ poms: pom });
    this.setState({ index });
  }

  render() {
    const { poms } = this.props;
    const { index } = this.state;
    return (
      <div>
        <button className="button gallery-btn" onClick={() => (
          this.changePom(-1)
        )}>
          Previous
        </button>
        <GalleryImage
          poms={poms[index]}
          onDelete={() => this.props.handleDelete(poms[index]._id, index)}
        />
        <button className="button gallery-btn" onClick={() => (
          this.changePom(+1)
        )}>
          Next
        </button>
      </div>
    );
  }
}