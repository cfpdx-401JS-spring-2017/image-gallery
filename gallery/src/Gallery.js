import React, { Component } from 'react';
import './Gallery.css';

class Gallery extends Component {
  constructor() {
    super();
    this.state ={
        galleryView: false,
        index: 0
    };
    this.showGallery = this.showGallery.bind(this);
    this.scrollNext = this.scrollNext.bind(this);
    this.scrollLast = this.scrollLast.bind(this);
  }
  
  showGallery() {
      let { galleryView } = this.state;
      galleryView = galleryView ? false : true;
      this.setState({ galleryView: galleryView })
  }

  scrollNext(){
      let { index } = this.state;
      if (index === 2) {
          this.setState({ index: 0 });
      } else {
          index += 1;
          this.setState({ index });
      }
  }

  scrollLast(){
      let { index } = this.state;
      if (index === 0) {
          this.setState({ index: 2 });
      } else {
          index -= 1;
          this.setState({ index: index});
      }
  }

  render() {
      let { galleryView, index } = this.state;
      let {images} = this.props;
      let divvy = null;

      if (galleryView) {
         divvy = <div className="YoungDivvy" >
            <p className="exitButton" onClick={this.showGallery}>exit</p>
            <p className="arrows" onClick={this.scrollLast}> {"<"} </p>
                <div className="WhoDaNeighbors">
                    <h3 className="ImageTitle">{images[index].title}</h3>
                    <img src={images[index].url} alt=""/>
                    <h5>{images[index].description}</h5>
                </div>
            <p className="arrows" onClick={this.scrollNext}> {">"} </p>
         </div>;
      } else {
          divvy = null;
      }
    return (
      <div className="Gallery">
       <button style={{
           textDecoration: 'none',
           backgroundColor: 'white',
           color: 'black',
           border: 'solid black 5px',
           margin: '2vh',
           width: '35vw',
           height: '8vh',
           fontSize: '3em'
       }}   onClick={this.showGallery}>Go To Gallery</button>
       {divvy}
      </div>  
    );
  }
}

export default Gallery;