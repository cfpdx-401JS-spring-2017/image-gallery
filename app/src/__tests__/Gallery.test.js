import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Gallery from '../Gallery';

const images = [
  {
    title: 'Black & Tan',
    description: 'Seriously heart-melting black and tan bunny... OMG <3',
    url: 'http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg'
  },
  {
    title: 'Falling Bunny',
    description: 'I don\'t know if this bunny fell and the picture was taken on impact, but I certainly prefer to think of it that way',
    url: 'http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg'
  },
  {
    title: 'Marshmellow Puff',
    description: 'Is it a rodent or a rabbit? I had to look it up to be sure. Then again, I\'m still not sure',
    url: 'http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg'
  }
];

describe('Gallery', () => {
  it('renders a Gallery view for Black & Tan', () => {
    const wrapper = shallow(
      <Gallery data={images} />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});