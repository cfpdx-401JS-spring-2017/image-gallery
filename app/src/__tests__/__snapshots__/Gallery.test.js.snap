// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery renders a Gallery view for Black & Tan 1`] = `
<div
  className="Gallery"
>
  <div
    className="button-container"
  >
    <button
      onClick={[Function]}
    >
      ←
    </button>
    <button
      onClick={[Function]}
    >
      →
    </button>
  </div>
  <div
    className="Gallery-view"
  >
    <h2>
      Black & Tan
    </h2>
    <p>
      Seriously heart-melting black and tan bunny... OMG &lt;3
    </p>
    <img
      alt="Black & Tan"
      className="Gallery-image"
      src="http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg"
    />
  </div>
</div>
`;
